### DISCUSSION TITLE
Stuck "Checking for updates..."

### DISCUSSION THREAD

**Discussion by ripcityhandyman** [pgale/lineage_15.1_Installation_TSV#11] [Q&A] [answered]
First off, thanks for the thorough instructions!

I have seen the notes mentioning the setup getting stuck when updating. I am running into that issue (stuck checking for updates for ~30 minutes) and I have retried the process 10+ times (powering off and back on again) with no luck. Has anyone else run into this and have any other steps to try?

**Comment by pgale** [comment]
What have you flashed from? A brand new device or something else?

You could try flashing the RSA image from Lenovo and try Lineage again after.

**Reply by ripcityhandyman** [reply, marked as answer]
@pgale, thanks for the response. I hadn't tried the RSA tool yet so I went ahead and gave it a shot. Unfortunately, same result.

But... I ended up getting my 3 TSVs successfully flashed! I had been using the EDL LiveCD to flash the Lineage image. Since I had Windows booted up for the RSA tool, I went ahead and installed the Windows tools and tried that method. And this method worked flawlessly :grinning: 

I haven't gone back to the Linux process to verify, but I noticed while following the Windows instructions that the ordering of the recovery phase is different. The Linux instructions say to factory clear _after_ sideloading Lineage and OpenGapps, whereas the Windows instructions say to factory clear _before_ sideloading. This may be the critical detail.

One other note that you may want to add to the instructions: After connecting to WiFi on the initial boot of Lineage, "Checking for updates" appears. On all three of my copies, the setup process hangs on this step indefinitely. The trick I found was to touch the "back" button after a few seconds. You'll actually see another "Checking for updates" page slide across from the left edge, on top of the existing "Checking for updates" page. It's at this point where touching "back" will allow the setup process to continue. If touching "back" returns you to the network list, then you didn't wait long enough.

Thanks again for all this documentation. Lineage has all the features I was missing from the original Android 8.1 build I was using, most importantly a functional screen timeout. And it anecdotally seems to be a bit snappier as well!

**Reply by pgale** [reply]
Thanks for that.

As far as I understand from reading about one million and one posts, doing a factory reset doesn't matter when you do it before the device is actually started into the OS. It's just clearing cache and user data which isn't set at that stage until it boots properly.

Re the checking for updates bit, I did see someone else mention that - your instructions are very useful so I will add that to the guide :) thanks!
